coord/src/rmr/rq.c

```diff
@@ -133,9 +137,26 @@ void RQ_Free(MRWorkQueue *q) {
   }
 
   uv_mutex_unlock(&q->lock);
+}
+
+static void RQ_Deactivate(uv_handle_t *handle) {
+  RedisModule_Log(NULL, "debug", "RQ_Deactivate: called");
+  MRWorkQueue *q = handle->data;
+  q->isActive = false;
+}
+
+void RQ_uvClose(MRWorkQueue *q) {
+  RedisModule_Assert(q->sz == 0);
+  uv_close((uv_handle_t *)&q->async, RQ_Deactivate);
+}
+
+void RQ_Free(MRWorkQueue *q, bool wait_for_uv_close) {
+  if (wait_for_uv_close) {
```

Comment:
Do we ever call it without `wait_for_uv_close`?

Reply:
No, also only on shutdown event with RS_GLOBAL_DTORS=1. but it should prevent whoever decides to use it in the future from misusing it and getting stuck in an infinite loop.

Reply:
I think its enough to document it. Maybe we can also make this function static? So no one will be able to call it from outside the rq unit (if we can call it unit ...)?

Reply:
RQ_Free must be called from outside the uv event loop after uv_close was called from the background, and it must be in that specific order. IMO adding an argument reduces this risk at a minimal cost, especially since it's not used in production.

It is called from rmr.c so must be exposed.

Reply:
I think its enough to document it, but your call.